### PR TITLE
Updated modals to use new UserFeedbackException

### PR DESCRIPTION
**Pull request by nick-prat**

### CODE REVIEW

File: src/foam/comics/v2/DAOCreateView.js [hunk 1]
```diff
@@ -112,31 +112,16 @@ foam.CLASS({
...
-          //     type: 'error'
-          //   }));
-          // }
-
-          if ( e.message === "An approval request has been sent out." ){
-            this.ctrl.add(this.NotificationMessage.create({
-              message: e.message,
-              type: 'info'
-            }));
+          if ( e.foamException && foam.comics.v2.userfeedback.UserFeedbackException.isInstance(e.foamException) && e.foamException.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/comics/v2/DAOCreateView.js]:
```suggestion
          if ( e.foamException && e.foamException.userFeedback  ) {
```
Would this suffice?

File: src/foam/comics/v2/DAOUpdateView.js [hunk 2]
```diff
@@ -130,31 +130,16 @@ foam.CLASS({
...
-          //     type: 'error'
-          //   }));
-          // }
-
-          if ( e.message === 'An approval request has been sent out.' ) {
-            this.ctrl.add(this.NotificationMessage.create({
-              message: e.message,
-              type: 'info'
-            }));
+          if ( e.foamException && foam.comics.v2.userfeedback.UserFeedbackException.isInstance(e.foamException) && e.foamException.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/comics/v2/DAOUpdateView.js]:
```suggestion
          if ( e.foamException &&  e.foamException.userFeedback  ) {
```

File: src/foam/comics/v2/DAOUpdateView.js [hunk 3]
```diff
@@ -130,31 +130,16 @@ foam.CLASS({
...
+            var currentFeedback = e.foamException.userFeedback;
+            while ( currentFeedback ){
```
Comment (jlhughes) [inline src/foam/comics/v2/DAOUpdateView.js]:
```suggestion
            while ( currentFeedback ) {
```

File: src/foam/box/RemoteException.js [hunk 4]
```diff
@@ -16,6 +16,10 @@ foam.CLASS({
     {
       class: 'String',
       name: 'message'
+    },
+    {
+      class: 'Object',
+      name: 'foamException'
```
Comment (kgrgreer) [inline src/foam/box/RemoteException.js]: Why not just call Exception?

File: src/foam/nanos/notification/NotificationRowView.js [hunk 5]
```diff
@@ -147,31 +147,18 @@
...
-          //     type: 'error'
-          //   }));
-          // }
-
-          if ( e.message === 'An approval request has been sent out.' ) {
-            self.ctrl.add(self.NotificationMessage.create({
-              message: e.message,
-              type: 'info'
-            }));
+          if ( e.foamException && foam.comics.v2.userfeedback.UserFeedbackException.isInstance(e.foamException) && e.foamException.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/nanos/notification/NotificationRowView.js]:
```suggestion
          if ( e.foamException && e.foamException.userFeedback  ) {
```

File: src/foam/nanos/notification/NotificationRowView.js [hunk 6]
```diff
@@ -147,31 +147,18 @@
...
+            var currentFeedback = e.foamException.userFeedback;
+            while ( currentFeedback ){
```
Comment (jlhughes) [inline src/foam/nanos/notification/NotificationRowView.js]:
```suggestion
            while ( currentFeedback ) {
```

File: src/foam/u2/DeleteModal.js [hunk 7]
```diff
@@ -108,25 +108,16 @@ foam.CLASS({
...
-
-          //     currentFeedback = currentFeedback.next;
-          //   }
-          // } else {
-          //   this.notify(err.message || this.FAIL_MSG, 'error');
-          // }
-
-          if ( err.message === "An approval request has been sent out."  ){
-            this.notify(err.message);
+          if ( err.foamException && foam.comics.v2.userfeedback.UserFeedbackException.isInstance(err.foamException) && err.FoamException.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/u2/DeleteModal.js]:
```suggestion
          if ( err.foamException && err.FoamException.userFeedback  ) {
```

File: src/foam/u2/DeleteModal.js [hunk 8]
```diff
@@ -108,25 +108,16 @@ foam.CLASS({
...
+            var currentFeedback = err.foamException.userFeedback;
+            while ( currentFeedback ){
```
Comment (jlhughes) [inline src/foam/u2/DeleteModal.js]:
```suggestion
            while ( currentFeedback ) {
```

File: src/foam/comics/v2/DAOCreateView.js [hunk 9]
```diff
@@ -112,31 +112,16 @@ foam.CLASS({
...
+            var currentFeedback = e.foamException.userFeedback;
+            while ( currentFeedback ){
```
Comment (jlhughes) [inline src/foam/comics/v2/DAOCreateView.js]:
```suggestion
            while ( currentFeedback ) {
```

**Comment by jlhughes** [comment]
@kgrgreer re: package - perhaps. 'foam.core.'  similar to java.lang   Now would be the time to fix it.

File: src/foam/comics/v2/DAOCreateView.js [hunk 10]
```diff
@@ -112,31 +112,16 @@ foam.CLASS({
...
-          //     type: 'error'
-          //   }));
-          // }
-
-          if ( e.message === "An approval request has been sent out." ){
-            this.ctrl.add(this.NotificationMessage.create({
-              message: e.message,
-              type: 'info'
-            }));
+          if ( e.exception && e.exception.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/comics/v2/DAOCreateView.js]:
```suggestion
          if ( e.exception && e.exception.userFeedback  ) {
```

File: src/foam/comics/v2/DAOUpdateView.js [hunk 11]
```diff
@@ -130,31 +130,16 @@ foam.CLASS({
...
-          //     type: 'error'
-          //   }));
-          // }
-
-          if ( e.message === 'An approval request has been sent out.' ) {
-            this.ctrl.add(this.NotificationMessage.create({
-              message: e.message,
-              type: 'info'
-            }));
+          if ( e.exception && e.exception.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/comics/v2/DAOUpdateView.js]:
```suggestion
          if ( e.exception && e.exception.userFeedback  ) {
```

File: src/foam/nanos/notification/NotificationRowView.js [hunk 12]
```diff
@@ -147,31 +147,18 @@
...
-          //     type: 'error'
-          //   }));
-          // }
-
-          if ( e.message === 'An approval request has been sent out.' ) {
-            self.ctrl.add(self.NotificationMessage.create({
-              message: e.message,
-              type: 'info'
-            }));
+          if ( e.exception && e.exception.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/nanos/notification/NotificationRowView.js]:
```suggestion
          if ( e.exception && e.exception.userFeedback  ) {
```

File: src/foam/u2/DeleteModal.js [hunk 13]
```diff
@@ -108,25 +108,16 @@ foam.CLASS({
...
-
-          //     currentFeedback = currentFeedback.next;
-          //   }
-          // } else {
-          //   this.notify(err.message || this.FAIL_MSG, 'error');
-          // }
-
-          if ( err.message === "An approval request has been sent out."  ){
-            this.notify(err.message);
+          if ( err.exception && err.exception.userFeedback  ){
```
Comment (jlhughes) [inline src/foam/u2/DeleteModal.js]:
```suggestion
          if ( err.exception && err.exception.userFeedback  ) {
```

**Pull request merged**